plugins/inputs/kube_state/configmap.go

```diff
@@ -0,0 +1,48 @@
+package kube_state
+
+import (
+	"context"
+	"time"
+
+	"github.com/ericchiang/k8s/apis/core/v1"
+
+	"github.com/influxdata/telegraf"
+)
+
+func collectConfigMaps(ctx context.Context, acc telegraf.Accumulator, ks *KubernetesState) {
+	list, err := ks.client.getConfigMaps(ctx)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	for _, s := range list.Items {
+		if s != nil {
+			if err = ks.gatherConfigMap(*s, acc); err != nil {
+				acc.AddError(err)
+				return
+			}
+		}
+	}
+}
+
+func (ks *KubernetesState) gatherConfigMap(s v1.ConfigMap, acc telegraf.Accumulator) error {
+	if s.Metadata.CreationTimestamp.GetSeconds() == 0 {
+		return nil
+	} else if !ks.firstTimeGather &&
+		ks.MaxConfigMapAge.Duration < time.Now().Sub(time.Unix(s.Metadata.CreationTimestamp.GetSeconds(), int64(s.Metadata.CreationTimestamp.GetNanos()))) {
+		return nil
+	}
+
+	fields := map[string]interface{}{
+		"gauge": 1,
+	}
+	tags := map[string]string{
+		"configmap_name":   s.Metadata.GetName(),
+		"namespace":        s.Metadata.GetNamespace(),
+		"resource_version": s.Metadata.GetResourceVersion(),
+	}
+
+	acc.AddFields(configMapMeasurement, fields, tags, time.Unix(s.Metadata.CreationTimestamp.GetSeconds(), int64(s.Metadata.CreationTimestamp.GetNanos())))
```

Comment:
Can do, but the "created at" as the timestamp was what was requested.

Reply:
I think it will be hard for this plugin to do "events" style metrics because instead of being event driven, this plugin uses polling.  This means that a simple implementation will report the same event each interval using the past timestamp for a long period of time, which is not very friendly to the time series database because it has to keep historical data open for writes.

Reply:
true. I think that's another reason this drug on, and "resources" aren't getting monitored this pass; those "events" style resources didn't seem to be useful when polling